Source/missiles.cpp

```diff
@@ -2810,69 +2810,66 @@ int AddMissile(Point src, Point dst, Direction midir, missile_id mitype, mienemy
 
 void MI_LArrow(Missile &missile)
 {
-	missile._mirange--;
+	int mind;
+	int maxd;
+	int eMind;
+	int eMaxd;
 	int p = missile._misource;
+	missile_graphic_id eAnim;
+	missile_resistance eRst;
+	auto &player = Players[p];

```

Comment:
Try to see if you can move these declarations closer to where they're used. In particular, `Players[p]` might be accessing OOB memory because `p` can be `-1`.